### ISSUE TITLE
Automatic deployment isn't working.

### ISSUE THREAD

**Issue by dblodgett-usgs** [cf-convention/cf-conventions#177]
@dopplershift -- it looks like the automatic deployment has been broken for some time. I don't have access to the cf-bot. Can you make sense of how this is supposed to work?

**Comment by marqh** [comment]
hello @dblodgett-usgs 

an initial look over 
https://github.com/cf-convention/cf-conventions/blob/master/.travis.yml
suggests that multiple failures have accreted over time.

the build is now working, e.g.
https://travis-ci.org/cf-convention/cf-conventions/builds/560188574
but I can't see where the build results are being sent.

I don't have access to `cf-conventions-bot`, afaik

As well as the ruby error which you fixed, there seems to be a disconnect regarding which repositories and branches are being targeted.

The cf website is being served by https://github.com/cf-convention/cf-convention.github.io, as far as I can tell.

However, the travis build never accesses this repository.
Instead, the travis build checks out the `gh-pages` branch from the cf-conventions project (I think)
https://github.com/cf-convention/cf-conventions/blob/master/.travis.yml#L24

It looks to me like it then tries to push the changes to the gh-pages branch of the cf-conventions repository:
https://github.com/cf-convention/cf-conventions/blob/master/.travis.yml#L41

This step succeeds, according to travis
https://travis-ci.org/cf-convention/cf-conventions/builds/560188574#L275

But, here's where the trail runs cold for me: 
https://github.com/cf-convention/cf-conventions/tree/gh-pages
hasn't been updated since August 2017
(co-incidentally automatically from the merger of my pull request https://github.com/cf-convention/cf-conventions/commit/679e3ed551f1b261b2d788a3e676b01c753143e6)
So, why has travis reported a successful push to gh-pages, but the repo hasn't picked it up.  I don't know.

It seems clear to me that the travis build is not communicating with the 
https://github.com/cf-convention/cf-convention.github.io
repository.  Perhaps this repo was created after the travis and cf bot were implemented.

However, it is far from clear to me how
http://cfconventions.org/latest.html
is serving 
http://cfconventions.org/cf-conventions/cf-conventions.html
If this is a github.io served website, then where in 
https://github.com/cf-convention/cf-convention.github.io
is cf-conventions/cf-conventions.html
I can't find it, and yet the URL:
http://cfconventions.org/cf-conventions/cf-conventions.html
resolves and provides a resource.
Understanding this is crucial, in my view, before anyone adapts a travis build, as I can't yet see what the target is supposed to be for a build.

To summarise: whilst I can see some potential issues with the .travis.yml, these do not explain the configuration and intent behind
http://cfconventions.org

I think finding a clueful person who is content to be seen as webmaster for this core resource is crucial here.
Finding out who has the keys to 
https://github.com/cf-conventions-bot
would also be useful

**Comment by DocOtak** [comment]
I can address the confusion with the urls and where things are coming from, there are three things interacting to create the behavior you are seeing:
* A custom domain for github pages (CNAME file in the cf-convention.gihub.io repo)
* An Organization page (The entire cf-convention.github.io repo)
* A Project Page (the gh-pages branch of the cf-convention repo)

The way github pages is set up defined that the repository named <orgname>.github.io will be the "organization page". I think this part is pretty understood based on the discussion above. You can also place an optional CNAME file (and do some DNS stuff), which will let you use a custom domain rather than "orgname.github.io". If you use a custom domain, you'll need to mentally replace where it says "username.github.io" or "orgname.github.io" while reading the documentation.

The next thing going on is that multiple repositories in an organization (or user account) can have github pages available. The way github pages handles this situation is by placing anything that is _not_ the organization page in a sub "directory" in the form of `orgname.github.io/<repo_name>`. In the github pages documentation, they call these "project pages". Which is why http://cfconventions.org/cf-conventions/index.html resolves to something that is not in the cf-convention.github.io repository.

Here is a demo with my personal stuff:
Here is my "personal website" I haven't updated for 6 years (since college) http://www.andrewbarna.org/ The source is here https://github.com/DocOtak/docotak.github.io

However, I did some mapping tests I wanted to show some folks https://github.com/DocOtak/maping_tests to show this off, I just enabled github pages for the master branch and it was automatically served at http://www.andrewbarna.org/maping_tests/tiling.html

I don't know what happens if there is a directory or asset in the organization repository which has the same name as a project repository.

**Comment by DocOtak** [comment]
More information:

Travis will ignore the exit codes of `after_success` so it might be failing silently see https://docs.travis-ci.com/user/job-lifecycle/#breaking-the-build

Travis supports an explicit "deployment" step to github pages, see https://docs.travis-ci.com/user/deployment/pages/ we should probably switch to this if possible

That last line of the `after_success` part has some stdout and stderr redirection, I don't know where git would print its messages but we might be missing some messages.

**Comment by dblodgett-usgs** [comment]
@marqh and @DocOtak -- your analysis is spot on by my understanding. 

The latest version of the spec comes from here: http://cfconventions.org/cf-conventions/ which is coming from here: https://github.com/cf-convention/cf-conventions/tree/gh-pages

The content of this page: http://cfconventions.org/ is coming from https://github.com/cf-convention/cf-convention.github.io

The `cf-convention` part of the path `http://cfconventions.org/cf-conventions/cf-conventions.html` refers to this repository.

I think the git push here: https://github.com/cf-convention/cf-conventions/blob/master/.travis.yml#L41 may actually be failing but since the output is piped to dev/null we aren't seeing it. This should be working as long as that github token works. I'll look into that a bit and see what we see.

**Comment by marqh** [comment]
> I think the git push here: https://github.com/cf-convention/cf-conventions/blob/master/.travis.yml#L41 may actually be failing but since the output is piped to dev/null we aren't seeing it. This should be working as long as that github token works. I'll look into that a bit and see what we see.

i think that sending this content to /dev/null is a security recommendation

if this is removed, tokens may be put into travis logs in plain text for all to see

so, if we need to do this to debug the probem, we may also need to revoke tokens as compromised and re-create them

**Comment by dblodgett-usgs** [comment]
@DocOtak Thanks for the link to travis deployment instructions. If someone else with permissions wants to grok that and set it up, cool. Otherwise I'll get to it when I get a chance.
